### DISCUSSION TITLE
Amazing project, why so little traction?

### DISCUSSION THREAD

**Discussion by maxgurewitz** [css-hooks/css-hooks#213] [General]
Hi there! This is an amazing project! It seems to solve the problem of:

1. runtime css-in-js libraries are not compatible with react server components
2. build time styling frameworks make assumptions about the consumer's build system, and are thus not as composable

In contrast, your project is purely runtime, not requiring a build time step, yet is compatible with react server components. What's the catch? Why does this project have so little attention?

**Comment by nsaunders** [comment]
Hi @maxgurewitz, thanks for your kind feedback!

I think you correctly identified the main advantages of this approach.

As for the "catch", a couple of limitations are explained on the [FAQ](https://css-hooks.com/docs/faq/). Aside than those, no one has brought any significant issues to my attention.

You may be interested to know that I'm working on v3 at the moment, which will be simpler and indeed more composable.

v2:
```jsx
<button
  style={css({
    color: "black",
    on: ($, { or }) => [
      $(or("&:hover", "&:focus"), {
        color: "blue"
      }),
      $("&:active", {
        color: "red"
      }),
      $("&:disabled", {
        color: "gray"
      })
    ]
  })}>
  Click me
</button>
```

v3:
```jsx
<button
  style={pipe(
    {
      color: "black"
    },
    on(or("&:hover", "&:focus"), {
      color: "blue"
    }),
    on("&:active", {
      color: "red"
    }),
    on("&:disabled", {
      color: "gray"
    })
  )}>
  Click me
</button>
```
v3 - pipeline operator ([someday?](https://github.com/tc39/proposal-pipeline-operator)):
```jsx
<button
  style={
    {
      color: "black"
    }
    |> on(or("&:hover", "&:focus"), {
      color: "blue"
    })
    |> on("&:active", {
      color: "red"
    })
    |> on("&:disabled", {
      color: "gray"
    })
  }>
  Click me
</button>
```

**Reply by maxgurewitz** [reply]
Hopefully, one day the pipeline operator is accepted!
